src/cargo/util/rlimit.rs

```diff
@@ -0,0 +1,106 @@
+#[cfg(unix)]
+use anyhow::bail;
+#[cfg(unix)]
+use libc::{RLIMIT_NOFILE, getrlimit, rlimit, setrlimit};
+
+use crate::CargoResult;
+
+pub struct ResourceLimits {
+    pub soft_limit: u64,
+    pub hard_limit: u64,
+}
+
+#[cfg(unix)]
+pub fn get_max_file_descriptors() -> CargoResult<ResourceLimits> {
+    let mut rlim = rlimit {
+        rlim_cur: 0,
+        rlim_max: 0,
+    };
+
+    let result = unsafe { getrlimit(RLIMIT_NOFILE, &mut rlim) };
+    if result != 0 {
+        bail!("Failed to get rlimit info with error code {result}")
+    }
```

Comment:
This is printing out the return value but the man page I was looking at said it is only ever `0` or `-1` and on `-1` you need to check `errno`